datanode/src/storage_api.py

```diff
@@ -380,6 +405,149 @@ def _DeleteGridCellMetaData(zoom, x, y, uss_id):
+      in these grid cells.
+
+  Returns:
+    200 and a new composite token if updated successfully,
+    409 if there is a locking conflict that could not be resolved, or
+    the other nominal 4xx error codes as necessary.
+  """
+  log.info('Grid cells metadata submit instantiated for %s at %sz, %s...',
+           uss_id, zoom, str(tiles))
+  sync_token = _GetRequestParameter('sync_token', None)
+  if not sync_token and 'sync_token' in request.headers:
+    sync_token = request.headers['sync_token']
+  scope = _GetRequestParameter('scope', None)
+  operation_endpoint = _GetRequestParameter('operation_endpoint', None)
+  operation_format = _GetRequestParameter('operation_format', None)
+  minimum_operation_timestamp = _GetRequestParameter(
+    'minimum_operation_timestamp', None)
+  maximum_operation_timestamp = _GetRequestParameter(
+    'maximum_operation_timestamp', None)
+  errorfield = errormsg = None
+  if not sync_token:
+    errorfield = 'sync_token'
+  elif not uss_id:
+    errorfield = 'uss_id'
+    errormsg = 'USS identifier not received from OAuth token check.'
+  elif not scope:
+    errorfield = 'scope'
+  elif not operation_endpoint:
+    errorfield = 'operation_endpoint'
+  elif not operation_format:
+    errorfield = 'operation_format'
+  elif not minimum_operation_timestamp:
+    errorfield = 'minimum_operation_timestamp'
+  elif not maximum_operation_timestamp:
+    errorfield = 'maximum_operation_timestamp'
+  if errorfield:
+    if not errormsg:
+      errormsg = errorfield + (
+        ' must be provided in the form data request to add to a '
+        'GridCell.')
+    result = {
+      'status': 'error',
+      'code': status.HTTP_400_BAD_REQUEST,
+      'message': errormsg
+    }
+  else:
+    result = wrapper.set_multi(zoom, tiles, sync_token, uss_id, scope,
+                         operation_format, operation_endpoint,
+                         minimum_operation_timestamp,
+                         maximum_operation_timestamp)
+  return result
+
+def _DeleteGridCellsMetaData(zoom, tiles, uss_id):
+  """Removes the USS entry in multiple GridCells.
+
+  Args:
+    zoom: zoom level in slippy tile format
+    tiles: array of x,y tiles to delete the uss from
+    uss_id: the plain text identifier for the USS from OAuth
+  Returns:
+    200 and a new sync_token if updated successfully,
+    409 if there is a locking conflict that could not be resolved, or
+    the other nominal 4xx error codes as necessary.
+  """
+  log.info('Grid cells metadata delete instantiated for %s, %sz, %s...',
+           uss_id, zoom, str(tiles))
+  if uss_id:
+    result = wrapper.delete_multi(uss_id, zoom, tiles)
+  else:
+    result = {
+      'status':
+        'fail',
+      'code':
+        status.HTTP_400_BAD_REQUEST,
+      'message':
+        """uss_id must be provided in the request to
+          delete a USS from a GridCell."""
+    }
+  return result
+
+
+def _ConvertCSVtoTiles(zoom):
```

Comment:
Maybe name this `_GetTilesFromRequest` since it doesn't actually accept a CSV to convert?